lib/action_subscriber/dsl.rb

```diff
@@ -79,5 +87,15 @@ def routing_key_for(method, routing_key_name)
     def routing_key_names
       @_routing_key_names ||= {}
     end
+
+    def run_action_with_filters(env, action)
+      subscriber_instance = self.new(env)
+      final_block = ->{ subscriber_instance.public_send(action) }
```

Comment:
They need to be procs so that I can have a closure around the subscriber instance. The scope of their existence is very limited so I'm not super concerned about causing memory leaks.

Also this is the same pattern that [protobuf](https://github.com/localshred/protobuf/blob/c0257ec32e4da9b3979dfd5dc6d554ff0c6bffe8/lib/protobuf/rpc/service_filters.rb#L212) uses for its around filters